ui/src/app/workflows/components/workflow-timeline/workflow-timeline.tsx

```diff
@@ -18,128 +18,124 @@ interface WorkflowTimelineProps {
     nodeClicked?: (node: models.NodeStatus) => any;
 }
 
-interface WorkflowTimelineState {
-    parentWidth: number;
-    now: moment.Moment;
-}
+export function WorkflowTimeline(props: WorkflowTimelineProps) {
+    let resizeSubscription: Subscription;
+    let refreshSubscription: Subscription;
 
-export class WorkflowTimeline extends React.Component<WorkflowTimelineProps, WorkflowTimelineState> {
-    private container: HTMLElement;
-    private resizeSubscription: Subscription;
-    private refreshSubscription: Subscription;
+    const [parentWidth, setParentWidth] = useState(0);
+    const [now, setNow] = useState(moment());
 
-    constructor(props: WorkflowTimelineProps) {
-        super(props);
-        this.state = {parentWidth: 0, now: moment()};
-        this.ensureRunningWorkflowRefreshing(props.workflow);
-    }
+    const containerRef = useRef(null);
 
-    public componentDidMount() {
-        this.resizeSubscription = fromEvent(window, 'resize').subscribe(() => this.updateWidth());
-        this.updateWidth();
-    }
+    useEffect(() => {
+        resizeSubscription = fromEvent(window, 'resize').subscribe(updateWidth);
+        updateWidth();
 
-    public componentWillReceiveProps(nextProps: WorkflowTimelineProps) {
-        this.ensureRunningWorkflowRefreshing(nextProps.workflow);
-    }
+        return () => {
+            if (resizeSubscription) {
+                resizeSubscription.unsubscribe();
+            }
+            if (refreshSubscription) {
+                refreshSubscription.unsubscribe();
+            }
```

Comment:
```suggestion
        return () => {
            resizeSubscription.current?.unsubscribe();
            refreshSubscription.current?.unsubscribe();
```

can simplify these with optional chaining syntax